src/main/main.ts

```diff
@@ -106,6 +103,12 @@ export type MixpanelInfo = {
 
 ipv4().then((value) => (ip = value));
 
+const mixpanelUUID = loadInstallerMixpanelUUID();
+const mixpanel: NineChroniclesMixpanel | undefined =
+  getConfig("Mixpanel") && !isDev
+    ? new NineChroniclesMixpanel(createMixpanel(MIXPANEL_TOKEN), mixpanelUUID, ip || "unknown")
```

Comment:
When we supplied `ip` for every track call it had no problem because eventually the value will be filled. However in this case the IP value may not be initialized by the time the NineChroniclesMixpanel instance is created (and I think it is likely to happen), and will keep uninitialized value for rest of runtime.

I'd suggest calling `ipv4` after intializing the instance, and call the setter to update the IP address.

Reply:
I realized `ip` is using only here. how about `ip` initialize on the `NineChroniclesMixpanel`?

Reply:
Yeah, that looks better for me.

Reply:
5a04499 amended.